src/i18n.ts

```diff
@@ -242,6 +242,50 @@ function validatePath(path: string): void {
 }
 
 /**
+ * Flatten an object containing nested objects into a single object with the keys joined by an underscore. Useful for passing
+ * objects to ICU message formatters, which do not support dot notation.
+ *
+ * E.g., `{ person: { name: 'Bill Evans' } }` becomes `{ 'person_name': 'Bill Evans' }`.
+ *
+ * @param options
+ * The format options object.
+ *
+ * @return
+ * The flattened format options object.
+ */
+export const flattenFormatOptions = (function () {
```

Comment:
It doesn't have to be, but those internal methods are not used by anything else, so it made sense to me to group them.